src/Transition/CreateNewIDFUsingRulesV22_2_0.f90

```diff
@@ -422,8 +424,57 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
               ! If your original object starts with R, insert the rules here
 
               ! If your original object starts with S, insert the rules here
-              
+
               ! If your original object starts with S, insert the rules here
+              CASE('SURFACEPROPERTY:LOCALENVIRONMENT')
+                CALL GetNewObjectDefInIDD(ObjectName,NwNumArgs,NwAorN,NwReqFld,NwObjMinFlds,NwFldNames,NwFldDefaults,NwFldUnits)
+                nodiff = .false.
+                matchedSurroundingName = MakeUPPERCase(InArgs(4))
+
+                ! Write out the original SurfaceProperty:LocalEnvironment object, with new F6
+                OutArgs = InArgs
+                OutArgs(6) = 'Gnd-'//matchedSurroundingName
+                CurArgs = CurArgs + 1
+                CALL WriteOutIDFLines(DifLfn,ObjectName,CurArgs,OutArgs,NwFldNames,NwFldUnits)
+                written = .true.
+
+                ! Now search for the child surrounding surfaces object to get required data
+                DO Num1=1,NumIDFRecords
+                  IF (MakeUPPERCase(IDFRecords(Num1)%Name) /= 'SURFACEPROPERTY:SURROUNDINGSURFACES') CYCLE
+                  IF (MakeUPPERCase(IDFRecords(Num1)%Alphas(1)) == matchedSurroundingName) THEN
+
+                    ! Alright, so get the necessary fields from the SurroundingSurfaces object
+                    ! Note that the surrounding surfaces object is updated separately
+                    SurroundingField1 = IDFRecords(Num1)%Numbers(2)
+                    SurroundingField2 = IDFRecords(Num1)%Alphas(3)
+
+                    ! Use the SurroundingSurfaces object saved fields F4 and F5 to create a new ground property object
+                    CALL GetNewObjectDefInIDD('SURFACEPROPERTY:GROUNDSURFACES',NwNumArgs,NwAorN,NwReqFld,NwObjMinFlds,NwFldNames,NwFldDefaults,NwFldUnits)
+                    OutArgs(1) = 'Gnd-'//matchedSurroundingName
+                    OutArgs(2) = 'Gnd-'//TRIM(matchedSurroundingName)//'-1'
+                    OutArgs(3) = SurroundingField1
+                    OutArgs(4) = SurroundingField2
+                    CurArgs = 4
+                    Call WriteOutIDFLines(DifLfn,'SurfaceProperty:GroundSurfaces',CurArgs,OutArgs,NwFldNames,NwFldUnits)
+
+                    EXIT
+                  ENDIF
+                ENDDO
```

Comment:
@Myoldmopar The transitioned objects look correct. I suggest one change, do nothing if the 4th field (`Surrounding Surfaces Object Name`) is blank. You may need to add an if block.

```

            IF (InArgs(4) /= Blank) THEN

                matchedSurroundingName = MakeUPPERCase(InArgs(4))
                ...
                DO Num1=1,NumIDFRecords
                ENDDO

            ENDIF
```

Reply:
Added the check, built and tested with the 22.1 file again and everything still seems fine.  Thanks @Nigusse